### PR TITLE
feat: Progress Stepper component

### PR DESCRIPTION
**Pull request by ramonmulia**
#### Progress Stepper component

added:

- react component
- tests
- docs
- html (macro)
- examples

Screenshot 2024-12-06 at 12 58 34

Screenshot 2024-12-06 at 12 58 46

Screenshot 2024-12-09 at 11 02 29

### CODE REVIEW

File: packages/design/tailwind/css/components.css [hunk 1]
```diff
@@ -944,6 +944,88 @@ input[type='file' i] {
...
+}
+
+.gi-progress-stepper[data-orientation='vertical'] {
+  @apply gi-flex-col;
+}
+
+.gi-progress-stepper-step {
+  @apply gi-text-white 
+  gi-rounded-full 
+  gi-relative
```
Comment (BogdanDanilescu) [inline packages/design/tailwind/css/components.css:960]: not sure if this is a formatting issue but we have some classes that are being displayed line by line and some that are being displayed one after another in a single line, we should decide which formatting style we want to follow so we are all aligned

**Review by BogdanDanilescu — APPROVED** [review]
LGTM, Great work on this 👍

File: packages/design/tailwind/css/components.css (hunk 1 above)
Comment (ramonmulia) [reply packages/design/tailwind/css/components.css:960]: good call, I believe it was something wrong with my lint plugin, going to put them in a line format.

File: packages/react/ds/src/progress-stepper/progress-stepper.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        className="gi-progress-stepper-step-container"
+        data-orientation={orientation}
+        data-current={isCurrentStep}
+        data-completed={isCompleted}
+        data-next={isNextStep}
+      >
+        <div className="gi-progress-stepper-step">
+          {isCompleted ? <Icon icon="check" /> : <div>{stepNumber}</div>}
+        </div>
+        {children}
```
Comment (pregno) [inline packages/react/ds/src/progress-stepper/progress-stepper.tsx:39]: Should we limit the number of chars?
Maybe let the content break on two lines if it exceeds an X number of chars and then ellipsis.

For example:
If you have a longer string like "This step is for requesting information."
we can render something like
```
This step is for
requesting info...
```

At least we don't risk to break the UI if the user put a longer text.

Comment (ramonmulia) [reply packages/react/ds/src/progress-stepper/progress-stepper.tsx:39]: good call, it makes sense, working on this.

**Review by pregno — APPROVED** [review]

File: packages/html/ds/src/progress-stepper/progress-stepper.test.ts
Comment (hamza14khan) [inline packages/html/ds/src/progress-stepper/progress-stepper.test.ts:1]: Don't you think we should be adding more tests? 😄

File: packages/react/ds/src/progress-stepper/types.ts
[the remark this answers is not in the document]
Comment (pregno) [reply packages/react/ds/src/progress-stepper/types.ts:1]: I don't see a possible use case; I thought it was informative.
If the user can navigate backwards and forward, it's preferable to have some navigation buttons, IMHO.

What use case do you see for a callback?

[the remark this answers is not in the document]
Comment (hamza14khan) [reply packages/react/ds/src/progress-stepper/types.ts:1]: Oh ok, additional navigation buttons will help.

Just a thought if a user wants to directly navigate from 3 to 5. Example a tutorial app, where step 1 is a course video and then second step is a quiz and then the last step is a survey. The user goes directly to the last step after completing the course video, skipping the quiz.

File: packages/html/ds/src/progress-stepper/progress-stepper.test.ts
Comment (ramonmulia) [reply packages/html/ds/src/progress-stepper/progress-stepper.test.ts:1]: yes for sure, I missed some.

**Pull request merged**